artemis-server/src/main/java/org/apache/activemq/artemis/core/postoffice/impl/DuplicateIDCacheImpl.java

```diff
@@ -69,11 +93,24 @@ public DuplicateIDCacheImpl(final SimpleString address,
 
       ids = new ArrayList<>(size);
 
+      cachedBoxedInts = boxedInts(size);
```

Comment:
Is this necessary if  -XX:AutoBoxCacheMax=Integer.Max and Integer.valueOf is used?

Reply:
IBM JVMs doesn't had it IIRC. Ideally Integer shouldn't be necessary...
I am preparing a further pr that won't use boxed types ;)

Reply:
nice

Reply:
I've implemented another version using a bespoke single-writer concurrent primitive valued hash map but...performance are a bit regressed (`ConcurrentHashMap` is still a nice piece of engineering, but I still need to understand why...)

And memory footprint isn't that great too (because of the small load-factor to save collisions): ~186 MB (that's similar to optimized boxed types)

Reply:
I've removed the last commit including the bespoke primitive map: seems that I cannot beat CHM (not a surprise) due to open-addressing nature of the version I've built. Big sized maps, almost full, work best as closed addressing and that's exactly the use case of CHM.